pkg/status/deployment.go

```diff
@@ -0,0 +1,78 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+	errs "github.com/pkg/errors"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrMsgCannotGetDeployment deployment not found
+	ErrMsgCannotGetDeployment = "unable to get the deployment"
+
+	// ErrMsgDeploymentConditionNotReady deployment not ready
+	ErrMsgDeploymentConditionNotReady = "deployment has unready status conditions"
+)
+
+// GetDeploymentStatusConditions looks up a deployment with the given name within the given namespace and checks its status
+// and finally returns a condition summarizing the status and an error if one occurred
+func GetDeploymentStatusConditions(client client.Client, name, namespace string) ([]toolchainv1alpha1.Condition, error) {
+	deploymentName := types.NamespacedName{Namespace: namespace, Name: name}
+	deployment := &appsv1.Deployment{}
+	err := client.Get(context.TODO(), deploymentName, deployment)
+	if err != nil {
+		err = errs.Wrap(err, ErrMsgCannotGetDeployment)
+		errCondition := NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentNotFoundReason, err.Error())
+		return []toolchainv1alpha1.Condition{*errCondition}, err
+	}
+
+	// get and check conditions
+	for _, condition := range deployment.Status.Conditions {
+		if (condition.Type == appsv1.DeploymentAvailable || condition.Type == appsv1.DeploymentProgressing) && condition.Status != corev1.ConditionTrue {
+			// there is a condition that is not ready, return it along with the error
+			errMsg := fmt.Sprintf("%s: %s", ErrMsgDeploymentConditionNotReady, condition.Type)
+			errCondition := NewComponentErrorCondition(toolchainv1alpha1.ToolchainStatusDeploymentNotReadyReason, errMsg)
+			return []toolchainv1alpha1.Condition{*errCondition}, fmt.Errorf(errMsg)
```

Comment:
you could get rid of the `fmt.Sprintf` call above since you already use `fmt.Errorf` here (but it's a minor thing)

Reply:
also, it's a bit weird that you return an error here. I was expecting an error if the client call failed, but not if one of the `DeploymentAvailable` or `DeploymentProgressing` conditions was not `true`. IMO, returning a specific `errCondition` may be enough here.

Reply:
>you could get rid of the fmt.Sprintf call above since you already use fmt.Errorf here (but it's a minor thing)
haha, true I'll change it

Reply:
>also, it's a bit weird that you return an error here

So for toolchain and member status I rely on the error to indicate a problem with a toolchain component. But, I understand what you mean. As I mentioned in https://github.com/codeready-toolchain/host-operator/pull/241#discussion_r453838062 maybe I could change it to a boolean instead. Would that be less confusing?

Reply:
My point is `[]toolchainv1alpha1.Condition{*errCondition}` already contains the error, so isn't it a bit of a duplicate if you return `fmt.Errorf(errMsg)` as well? I'm not sure if a `bool` is necessary. Maybe just checking the content of the conditions is enough?

Reply:
See comment below related to kubefed conditions

Reply:
I'm ok with addressing this later ;)